drivers/rtc/rtc_emul.c

```diff
@@ -245,27 +238,22 @@ static void rtc_emul_invoke_update_callback(const struct device *dev)
 }
 #endif /* CONFIG_RTC_UPDATE */
 
-static void rtc_emul_update(struct k_work *work)
+static void rtc_emul_update(struct k_timer *timer)
 {
-	struct rtc_emul_work_delayable *work_delayable = (struct rtc_emul_work_delayable *)work;
-	const struct device *dev = work_delayable->dev;
+	const struct device *dev = timer->user_data;
 	struct rtc_emul_data *data = (struct rtc_emul_data *)dev->data;
 
-	k_work_schedule(&work_delayable->dwork, K_MSEC(1000));
-
-	k_mutex_lock(&data->lock, K_FOREVER);
-
-	rtc_emul_increment_tm(&data->datetime);
+	K_SPINLOCK_GUARDED_SECTION(&data->lock) {
```

Comment:
I like this pattern but it would probably be better off as a separate tree-wide PR.

Reply:
Ignored for now as not being part of this PR - see discussion below.